nixos/modules/services/monitoring/prometheus/exporters/snmp.nix

```diff
@@ -81,6 +81,32 @@ in
         Only log messages with the given severity or above.
       '';
     };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/root/prometheus-snmp-exporter.env";
+      description = ''
+        Environment file as defined in {manpage}`systemd.exec(5)`.
+
+        Secrets may be passed to the service without adding them to the
+        world-readable Nix store, by specifying placeholder variables as
+        the option value in Nix and setting these variables accordingly in the
+        environment file.
+
+        Environment variables from this file will be interpolated into the
+        config file using envsubst with this syntax:
+        `$ENVIRONMENT ''${VARIABLE}`
+
+        For variables to use see [Prometheus Configuration](https://github.com/prometheus/snmp_exporter#prometheus-configuration).
+
+        Besides setting the filepath for this option, you also need to add
+        flag `--config.expand-environment-variables` to `extraFlags`.
+
```

Comment:
We should do that if the environmentFile is defined to make this option even more useful